astropy/modeling/functional_models.py

```diff
@@ -400,13 +400,15 @@ def evaluate(x, factor):
         return factor * x
 
 
-class Redshift(Fittable1DModel):
+class RedshiftWavelength(Fittable1DModel):
     """
     One dimensional redshift model.
 
+    Redshifts a wavelength array from rest (z=0) to the input redshift.
+
     Parameters
     ----------
-    z : float or a list of floats
+    z : int or float
```

Comment:
@pllim Yes, I agree that case is very useful, but I didn't see any obvious way to do that.  I see now that the "trick" is to use "model_set_axis":

```
wavelen = np.arange(10)
r = Redshift(z=[1, 2], model_set_axis=0)    # generates 2 models
r(wavelen, model_set_axis=False)
```
 Because this is a general feature of models, I'll defer to @embray as to how parameters should be documented, e.g. every model parameter could be array-like when using `model_set_axis` (although I'm not in favor that).  From a quick look in `functional_models`, `Redshift` seems to be the only model that currently does that.